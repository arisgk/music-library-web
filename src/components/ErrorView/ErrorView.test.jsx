/* global describe, beforeAll, it, expect */
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { createShallow } from '@material-ui/core/test-utils';
import NoteIcon from '@material-ui/icons/MusicNote';
import Wrapper, { ErrorView } from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Home', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow({ untilSelector: ErrorView });
  });

  it('renders message and icon', () => {
    const message = 'Something went wrong';
    const icon = <NoteIcon />;

    const wrapper = shallow(<Wrapper message={message} icon={icon} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders message, icon and button', () => {
    const message = 'Something went wrong';
    const icon = <NoteIcon />;
    const buttonLabel = 'Refresh';

    const wrapper = shallow(
      <Wrapper message={message} icon={icon} buttonLabel={buttonLabel} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
