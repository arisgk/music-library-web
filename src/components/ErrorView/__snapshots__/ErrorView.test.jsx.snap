// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders message and icon 1`] = `
<div
  className="ErrorView-container-1"
>
  <div
    className="ErrorView-iconContainer-2"
  >
    <pure(ErrorIcon)
      classes={
        Object {
          "root": "ErrorView-icon-3",
        }
      }
    />
  </div>
  <h3
    className="ErrorView-message-4"
  >
    Something went wrong
  </h3>
</div>
`;

exports[`Home renders message, icon and button 1`] = `
<div
  className="ErrorView-container-1"
>
  <div
    className="ErrorView-iconContainer-2"
  >
    <pure(ErrorIcon)
      classes={
        Object {
          "root": "ErrorView-icon-3",
        }
      }
    />
  </div>
  <h3
    className="ErrorView-message-4"
  >
    Something went wrong
  </h3>
  <WithStyles(Button)
    className="ErrorView-button-5"
    color="primary"
    onClick={[Function]}
    variant="contained"
  >
    Refresh
  </WithStyles(Button)>
</div>
`;
